CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+# Contributing to B2 Command Line Tool
+
+We encourage outside contributors to perform changes on our codebase. Many such changes have been merged already. In order to make it easier to contribute, core developers of this project:
+
+* provide guidance (through the issue reporting system)
+* provide tool assisted code review (through the Pull Request system)
+* maintain a set of unit tests
+* maintain a set of integration tests (run with a production cloud)
+* maintain development automation tools using [nox](https://github.com/theacodes/nox) that can easily:
+   * format the code using [yapf](https://github.com/google/yapf)
+   * runs linters to find subtle/potential issues with maintainability
+   * run the test suite on multiple Python versions using [pytest](https://github.com/pytest-dev/pytest)
+* maintain Continuous Integration (by using GitHub Actions) that:
+   * runs all sorts of linters
+   * checks if the Python distribution can be built
+   * runs all tests on a matrix of 6 versions of Python (including pypy) and 3 operating systems (Linux, Mac OS X and Windows)
+   * checks if the documentation can be built properly
+* maintain other Continuous Integration tools (coverage tracker)
+
+## Developer Info
+
+You'll need to have [nox](https://github.com/theacodes/nox) installed:
+
+* `pip install nox`
+
+With `nox`, you can run different sessions (default are `lint` and `test`):
+
+* `format` -> Format the code.
+* `lint` -> Run linters.
+* `test` (`test-3.5`, `test-3.6`, `test-3.7`, `test-3.8`) -> Run test suite.
+* `cover` -> Perform coverage analysis.
+* `build` -> Build the distribution.
+* `deploy` -> Deploy the distribution to the PyPi.
+* `doc` -> Build the documentation.
+* `doc_cover` -> Perform coverage analysis for the documentation.
+
+For example:
+
+    $ nox -s format
+    nox > Running session format
+    nox > Creating virtual environment (virtualenv) using python3.8 in .nox/format
+    ...
+
+    $ nox -s format
+    nox > Running session format
+    nox > Re-using existing virtual environment at .nox/format.
+    ...
+
+    $ nox --no-venv -s format
+    nox > Running session format
+    ...
+
+Sessions `test` ,`unit`, and `integration` can run on many Python versions, 3.5-3.8 by default.
+
+Sessions other than than use the last given Python version, 3.8 by default.
+
+You can change it:
+
+    export NOX_PYTHONS=3.6,3.7
+
+With the above setting, session `test` will run on Python 3.6 and 3.7, and all other sessions on Python 3.7.
+
+Given Python interpreters should be installed in the operating system or via [pyenv](https://github.com/pyenv/pyenv).
+
+## Linting
+
+To run all available linters:
+
+    nox -s lint
+
+We marked the places in the code which are significantly less intuitive than others in a special way. To find them occurrences, use `git grep '*magic*'`.
+
+## Testing
```

Comment:
is there no automated watcher for changes, so that documentation build/tests are done as soon as you save a file (via INOTIFY)? I used to have a script for that, it's now gone and instruction doesn't mention anything like it

Reply:
There was a script for docs and `nox -s doc` watches for changes (in docs src and python src) by default.
I don't remember something like that for running tests, but I will check again what was removed. I don't use that kind of features for testing, but I know that almost every IDE have something like that (including vim and emacs)